api/files.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import hashlib
+import logging
+import shutil
+import json
+import os
+
+from . import util
+
+log = logging.getLogger('scitran.api')
+
+
+class FileRequest(object):
+    """This class provides and interface for file uploads.
+    To perform an upload the client of the class should follow these steps:
+
+    1) initialize the request
+    2) save a temporary file
+    3) check identical
+    4) move the temporary file to its destination
+
+    The operations could be safely interleaved with other actions like permission checks or database updates.
+    """
+
+    def __init__(self, client_addr, filename, body, received_md5, metadata, tags):
```

Comment:
Let's use the same algorithm for both purposes. MD5 is obsolete for a variety of reasons